flink-core/src/main/java/org/apache/flink/configuration/IllegalConfigurationException.java

```diff
@@ -42,6 +42,17 @@ public IllegalConfigurationException(String message) {
 
 	/**
 	 * Constructs an new IllegalConfigurationException with the given error message
+	 * format and arguments.
+	 *
+	 * @param format The error message format for the exception.
+	 * @param arguments The arguments for the format.
+	 */
+	public IllegalConfigurationException(String format, Object... arguments) {
```

Comment:
Curious: Why introduce the extra constructor and not call `String.format(...)` where the exception is created?

Reply:
Convenience - reduces boiler plate. I prefer this:
```
throw new IllegalConfigurationException("%s %s", foo, bar);
```
to that:
```
throw new IllegalConfigurationException(String.format("%s %s", foo, bar));
```
same as it is done in loggers.